test/test_cache.py

```diff
@@ -0,0 +1,232 @@
+import os
```

Comment:
and why not to keep tests along side with the code base?  I do understand for web services etc - useless and possibly really undesired to ship tests along, but for generic libraries it is only of benefit to be able to test "as installed" (thus having tests shipped within module).  We have that (simpler imho) setup within datalad,dandi-cli etc.

Reply:
How often do you — or anyone — actually test packages that are already installed?  There isn't even any packaging functionality for indicating what command(s) to run to test an installed package.  Testing of code "as installed" should be done by the developer by installing the code and running the tests against the installed copy, and you don't need the tests folder to be under the package folder for that.

Reply:
> How often do you — or anyone — actually test packages that are already installed?

well, I do not think I do that often, but I do use `git-annex test`, more rarely `datalad test` (since we typically test in development - the curse of being its developer instead of a user) in particular when I install them on some new system -- it allows to test that the beast (as installed) operational on that system.  Also in DataLad Travis setup we came to that as well (run tests out of source, we cd to some directory) so we verify that "nothing is forgotten to be installed".  
With `pip install -e` (even with separation of `src/`) I believe you would just use the package under `src/PACKAGE`, so if something (data files etc) was forgotten to be installed -- you are out of luck.

I think for this package it might be actually quite important/convenient to have tests shipped along since we might discover all kinds of file systems quirks, and it would be easier just to instruct user to run shipped along tests (yet to figure out how since as you mention "packaging" is lacking uniform way to run tests).

> Testing of code "as installed" should be done by the developer

well, I would say it is "typically done" not necessarily "should [only] be done" by the developer.  I do remember rare occasions when I did run tests for some library as installed to make sure that everything is kosher in current installation. 

>  There isn't even any packaging functionality for indicating what command(s) to run to test an installed package.

`python -m nose PACKAGE` would run tests. I think `pytest` indeed would not pickup the package from installed location and needs path (correct me if I am wrong) for discovery.

So, overall, I do feel somewhat strongly about having tests shipped along (unless their require heavy auxiliary data files etc).

Reply:
Pytest can test an installed package with `pytest --pyargs PACKAGE`.

I've moved the tests under the package code.